platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/internal/AbstractTaskSchedulerBuilder.java

```diff
@@ -146,6 +165,7 @@ public AbstractTaskSchedulerBuilder<OUT> withType(@NonNull final TaskSchedulerTy
     @Override
     @NonNull
     public AbstractTaskSchedulerBuilder<OUT> withUnhandledTaskCapacity(final long unhandledTaskCapacity) {
+        validateDirectScheduler(this.type, unhandledTaskCapacity);
```

Comment:
I'd suggest you only do this check when the scheduler is built. There could be some weirdness where it is possible to get an error while entering a legal combination of configurations but in the wrong order.